geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -2081,10 +2081,12 @@ private void distributedUnlockForClear() {
   /**
    * obtain locks preventing generation of new versions in other members
    */
-  private void obtainWriteLocksForClear(RegionEventImpl regionEvent,
+  protected void obtainWriteLocksForClear(RegionEventImpl regionEvent,
       Set<InternalDistributedMember> participants) {
     lockLocallyForClear(getDistributionManager(), getMyId(), regionEvent);
-    DistributedClearOperation.lockAndFlushToOthers(regionEvent, participants);
+    if (!isUsedForPartitionedRegionBucket()) {
+      DistributedClearOperation.lockAndFlushToOthers(regionEvent, participants);
```

Comment:
Not part of this diff, but I assume we are taking a less granular lock earlier in the process for the entire partitioned region?

Reply:
BR does not need to send msg to secondary to do rvvLock.